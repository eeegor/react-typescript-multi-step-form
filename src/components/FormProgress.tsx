/**
 * Component FormProgress
 */

import * as React from 'react';
import * as classnames from 'classnames';
import './FormProgress.scss';

export interface Props {
	maxSteps: number;
	currentStep: number;
	className?: string;
}

export class FormProgress extends React.Component<Props> {
	render(): JSX.Element {
		const { className, maxSteps, currentStep } = this.props;

		return (
			<div className={classnames('form-progress', className)}>
				<div
					className="form-progress__indicator-bar"
					style={{ width: `${(100 / (maxSteps - 1)) * (currentStep - 1)}%` }}
				/>
				<div className="form-progress__background-bar" />
			</div>
		);
	}
}
