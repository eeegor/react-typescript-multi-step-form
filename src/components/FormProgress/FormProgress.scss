@import '../../variables';

.form-progress {
	position: relative;
	width: 100%;
	z-index: -1;
	height: 2rem;
	margin-left: 0.8rem;

	&__background-bar,
	&__indicator-bar {
		top: calc(50% - 2px);
		position: absolute;
		height: 4px;
	}

	&__indicator-bar {
		background: $color-primary;
		margin-right: 4px;
		z-index: 2;
		transition: width 0.5s cubic-bezier(0.075, 0.82, 0.165, 1);
	}

	&__background-bar {
		width: 100%;
		background: $color-grey-light;
		z-index: 1;
	}
}
