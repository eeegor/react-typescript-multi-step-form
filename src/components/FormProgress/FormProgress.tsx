/**
 * Component FormProgress
 */

import React from 'react';
import classnames from 'classnames';
import './FormProgress.scss';

export interface Props {
	maxSteps: number;
	currentStep: number;
	className?: string;
}

export const FormProgress = (props: Props): JSX.Element => {
	const { className, maxSteps, currentStep } = props;

	return (
		<div className={classnames('form-progress', className)}>
			<div
				className="form-progress__indicator-bar"
				style={{
					width: `${(100 / (maxSteps - 1)) * (currentStep - 1)}%`,
				}}
			/>
			<div className="form-progress__background-bar" />
		</div>
	);
};
