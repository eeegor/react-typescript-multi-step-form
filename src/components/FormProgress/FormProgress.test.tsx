import React from 'react';
import renderer from 'react-test-renderer';
import { FormProgress } from './FormProgress';

describe('FormProgress', () => {
	it('renders markup correctly', () => {
		const wrapper = <FormProgress maxSteps={3} currentStep={1} />;
		const tree = renderer.create(wrapper).toJSON();
		expect(tree).toMatchSnapshot();
	});
});
