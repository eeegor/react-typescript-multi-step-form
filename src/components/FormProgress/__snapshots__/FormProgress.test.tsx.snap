// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormProgress renders markup correctly 1`] = `
<div
  className="form-progress"
>
  <div
    className="form-progress__indicator-bar"
    style={
      Object {
        "width": "0%",
      }
    }
  />
  <div
    className="form-progress__background-bar"
  />
</div>
`;
